packages/grafana-prometheus/src/language_provider.ts

```diff
@@ -335,13 +343,22 @@ export default class PromQlLanguageProvider extends LanguageProvider {
    * @param name
    * @param withName
    */
-  fetchSeriesLabels = async (name: string, withName?: boolean): Promise<Record<string, string[]>> => {
+  fetchSeriesLabels = async (
```

Comment:
Since we are changing the signature of this function, I believe we should also update some places where we call this function. Maybe that's outside of the scope of this PR. If that's the case please let me know. This function is being called by `getSeries` and that is called by `getSeriesValues` and `getSeriesValues` and that is eventually used in `monaco-completion-provider`. So if there is no need for `monaco` we don't need to do anything but I guess it's worth a check. What do you think?

Reply:
I think it would be a good idea to write more about this in issues for the other areas to add the limit, but it is out of scope for this issue. I can add the other areas to this issue https://github.com/grafana/grafana/issues/91544

Reply:
And if it's called in other places, I will need to add the  `none` param. I will check this.

Reply:
Updated the code and updated the full issue to include the limit param to cover all the other cases https://github.com/grafana/grafana/issues/91544